jkube-kit/watcher/standard/src/main/java/org/eclipse/jkube/watcher/standard/DockerImageWatcher.java

```diff
@@ -160,21 +160,21 @@ private void updateImageName(KubernetesClient kubernetes, HasMetadata entity, St
             Deployment resource = (Deployment) entity;
             DeploymentSpec spec = resource.getSpec();
             if (spec != null && updateImageName(entity, spec.getTemplate(), imagePrefix, imageName)) {
-                kubernetes.apps().deployments().inNamespace(namespace).resource(resource).replace();
+                kubernetes.apps().deployments().inNamespace(namespace).resource(resource).update();
```

Comment:
In order to properly swap the deprecated `replace` call with `update` the resource should at least be unlocked to ensure that no optimistic locking breaks the current behavior:

```java
/* ... */.resource(resource).unlock().update()
```

Reply:
made the necessary changes.